open_humans/account_views.py

```diff
@@ -141,3 +174,149 @@ def delete(self, request, *args, **kwargs):
 
     def get_object(self, queryset=None):
         return self.request.user
+
+
+class ResetPasswordView(FormView):
+    """
+    Ooops, we've done lost our password, Martha!
+    """
+    template_name = 'account/password_reset.html'
+    form_class=ResetPasswordForm
+    success_url = reverse_lazy("account_reset_password_done")
+
+    def get_form_class(self):
+        return get_form_class(FORMS,
+                              'reset_password',
+                              self.form_class)
+
+    def form_valid(self, form):
+        form.save(self.request)
+        return super().form_valid(form)
+
+    def get_context_data(self, **kwargs):
+        ret = super().get_context_data(**kwargs)
+        login_url = passthrough_next_redirect_url(self.request,
+                                                  reverse("account_login"),
+                                                  'next')
+        # NOTE: For backwards compatibility
+        ret['password_reset_form'] = ret.get('form')
+        # (end NOTE)
+        ret.update({"login_url": login_url})
+        return ret
+
+    def post(self, request, *args, **kwargs):
+        request = super().post(request, *args, **kwargs)
+        return request
+
+
+class PasswordResetFromKeyView(FormView):
```

Comment:
Same question here - is there some reason not to inherit from the equivalent in `allauth`?

Reply:
In this case, allauth tries to do something overly clever that doesn't quite work.